crates/extrinsics/src/lib.rs

```diff
@@ -403,28 +455,28 @@ fn parse_contract_account_address(
 /// requested elements starting from an optional address
 pub async fn fetch_all_contracts(
     client: &Client,
+    rpc: RpcClient,
     count: u32,
-    count_from: Option<&AccountId32>,
 ) -> Result<Vec<AccountId32>> {
-    let key = api::storage()
-        .contracts()
-        .contract_info_of_root()
-        .to_root_bytes();
-    let start_key = count_from
-        .map(|e| [key.clone(), hashing::twox_64(&e.0).to_vec(), e.0.to_vec()].concat());
-    let keys = client
-        .storage()
-        .at_latest()
-        .await?
-        .fetch_keys(key.as_ref(), count, start_key.as_deref())
-        .await?;
+    let query = api::storage().contracts().contract_info_of_iter();
+    let root_key = query.to_root_bytes();
 
-    let contracts = keys
-        .into_iter()
-        .map(|e| parse_contract_account_address(&e.0, key.len()))
-        .collect::<Result<_, _>>()?;
+    let best_block = get_best_block(rpc).await?;
+    let mut contracts = client.storage().at(best_block).iter(query).await?;
+
+    let mut contract_accounts = Vec::new();
+    while let Some(result) = contracts.next().await {
```

Comment:
Is there a "bulk fetch" function, which we could use to speed up the execution?

Reply:
In the background it is doing pagination so every iteration does not result in a network request. https://github.com/paritytech/subxt/blob/cf7e2db1b7b57385d2e0bb0e49169bd6627f6811/subxt/src/backend/legacy/mod.rs#L342

Anyway since I switched it to `fetch_keys` it is now back down to 3s.